lib/pbench/agent/results.py

```diff
@@ -18,49 +18,55 @@ def __init__(self, result_dir, target_dir, user, prefix, config, logger):
         assert (
             config and logger
         ), f"config, '{config!r}', and/or logger, '{logger!r}', not provided"
+        self.result_dir = result_dir
+        self.target_dir = target_dir
+        self.user = user
+        self.prefix = prefix
+        self.config = config
+        self.logger = logger
+
+    def check_result_target_dir(self, result_dir, target_dir):
+        sts = 0
         if not result_dir:
-            logger.error("Result directory not provided")
-            sys.exit(1)
+            self.logger.error("Result directory not provided")
+            sts = 1
         if not target_dir:
-            logger.error("Target directory not provided")
-            sys.exit(1)
+            self.logger.error("Target directory not provided")
+            sts = 1
         try:
             self.result_dir = Path(result_dir).resolve(strict=True)
         except FileNotFoundError:
-            logger.error("Invalid result directory provided: {}", result_dir)
-            sys.exit(1)
+            self.logger.error("Invalid result directory provided: {}", result_dir)
+            sts = 1
         else:
             if not self.result_dir.is_dir():
-                logger.error("Invalid result directory provided: {}", result_dir)
-                sys.exit(1)
+                self.logger.error("Invalid result directory provided: {}", result_dir)
+                sts = 1
         try:
             self.target_dir = Path(target_dir).resolve(strict=True)
         except FileNotFoundError:
-            logger.error("Invalid target directory provided: {}", target_dir)
-            sys.exit(1)
+            self.logger.error("Invalid target directory provided: {}", target_dir)
+            sts = 1
         else:
             if not self.target_dir.is_dir():
-                logger.error("Invalid target directory provided: {}", target_dir)
-                sys.exit(1)
-        self.user = user
-        self.prefix = prefix
-        self.config = config
-        self.logger = logger
+                self.logger.error("Invalid target directory provided: {}", target_dir)
+                sts = 1
+        return sts
 
     def make_result_tb(self):
         if os.path.exists(f"{self.result_dir}.copied"):
             self.logger.debug("Already copied {}", self.result_dir)
-            sys.exit(0)
+            return (0, "")
 
         pbench_run_name = self.result_dir.name
         if os.path.exists(f"{self.result_dir}/.running"):
-            self.logger.debug(
+            self.logger.warning(
                 "The benchmark is still running in {} - skipping; if that is"
                 " not true, rmdir {}/.running, and try again",
                 pbench_run_name,
                 pbench_run_name,
             )
-            sys.exit(0)
+            return (0, "")
```

Comment:
Returning tuples is an awkward python idiom; it's often convenient, but it's not usually convenient to understand or use. At the very least, we need good docstrings describing exactly what's returned so one doesn't have to search through the code to figure out return types. We want to add type hints; but this also is no help with tuple return values as saying that the return is a "tuple" gives no hint what that means. It has to be explicitly documented in the docstring.

Again, I'd really prefer that we use exceptions with meaningful names and payloads to report errors; including, if we want that, integer status codes.

Reply:
took care of it.. :)